src/highdicom/seg/sop.py

```diff
@@ -860,8 +746,229 @@ def add_segments(
         if len(self.PixelData) % 2 == 1:
             self.PixelData += b'0'
 
+        self.copy_specimen_information(src_img)
+        self.copy_patient_and_study_information(src_img)
+
+        # Build lookup tables for efficient decoding
         self._build_luts()
 
+    def add_segments(
+        self,
+        pixel_array: np.ndarray,
+        segment_descriptions: Sequence[SegmentDescription],
+        plane_positions: Optional[Sequence[PlanePositionSequence]] = None,
+        omit_empty_frames: bool = True,
+    ) -> None:
+        """To ensure correctness of segmentation images, this
+        method was deprecated in highdicom 0.8.0. For more information
+        and migration instructions see :ref:`here <add-segments-deprecation>`.
+
+        """  # noqa E510
+        raise AttributeError(
+            'To ensure correctness of segmentation images, the add_segments '
+            'method was deprecated in highdicom 0.8.0. For more information '
+            'and migration instructions visit '
+            'https://highdicom.readthedocs.io/en/latest/release_notes.html'
+            '#deprecation-of-add-segments-method'
+        )
+
+    @staticmethod
+    def _check_segment_numbers(described_segment_numbers: np.ndarray):
+        """Checks on segment numbers extracted from the segment descriptions.
+
+        Segment numbers should start at 1 and increase by 1. This method checks
+        this and raises an appropriate exception for the user if the segment
+        numbers are incorrect.
+
+        Parameters
+        ----------
+        described_segment_numbers: np.ndarray
+            The segment numbers from the segment descriptions, in the order
+            they were passed. 1D array of integers.
+
+        Raises
+        ------
+        ValueError
+            If the described_segment_numbers do not have the required values
+
+        """
+        # Check segment numbers in the segment descriptions are
+        # monotonically increasing by 1
+        if not (np.diff(described_segment_numbers) == 1).all():
+            raise ValueError(
+                'Segment descriptions must be sorted by segment number '
+                'and monotonically increasing by 1.'
+            )
+        if described_segment_numbers[0] != 1:
+            raise ValueError(
+                'Segment descriptions should be numbered starting '
+                f'from 1. Found {described_segment_numbers[0]}. '
+            )
+
+    @staticmethod
+    def _check_pixel_array(
```

Comment:
This method does more than just checking the array. It also casts it to the correct data type if necessary. It may be beneficial for the name to reflect that.

```suggestion
    def _check_and_cast_pixel_array(
```

Reply:
resolved in fd1384d